wordpress-mobile/WordPress-Android
# RI: Choose Interests - Navigate to discover

Master Issue:  #12027

This PR attempts to show and close "Choose Interests" screen from the `ReaderFragment`. 

I couldn't find a scenario in the app where we navigate from 
_bottom nav tab fragment_ => _full screen fragment_ that remains above the bottom nav bar. 

Let me know if there is one.

Tried to use `WPMainNavigationView` to add `ReaderInterestsFragment` to backstack for Reader tab and pop it when Done button is clicked but finally used below approach as it appeared to be simpler:

   - Start with a hidden `AppBar` inside the `ReaderFragment` without any tabs 
   - Add `ReaderInterestsFragment` as a child fragment.
   - On Done button click inside the `ReaderInterestsFragment`, use the parent fragment's viewmodel (`ReadeViewModel`) to trigger these actions:
        - Remove `ReaderInterestsFragment` child fragment
        - Fetch reader viewpager tabs 
        - Unhide `AppBar` and start displaying the reader tabs

### To test

1. Launch the app
2. Go to Reader tab
3. Notice that  "Choose Interests" screen is shown
4. Choose some interests and press the Done button
5. Notice that  "Choose Interests" screen is closed and Discover tab is shown

----

6. Close the app and Repeat Steps 1 - 3
7. Notice that  "Choose Interests" screen is shown
8. Click device back button
9. App goes to background (as we were still on the `ReaderFragment` attached to the bottom nav bar)

#### Notes

 - The [condition to show the screen](https://github.com/wordpress-mobile/WordPress-Android/blob/9b86e7dba03bb333bfb48b8c2fafa9788902f6d5/WordPress/src/main/java/org/wordpress/android/ui/reader/viewmodels/ReaderViewModel.kt#L46)  is not final. It will depend on this separate task:
      - [ ] Fetch List of user's tags and show "Select Topics" screen only if the list is empty - get mocked list of tags from ReaderRepository

As we're still finalising the approach, pending tasks for this PR
- [X] Test app for config changes, bottom bar tab toggles etc 
- [ ] Add transition animations 

PR submission checklist:
